v2/workloadapi/client_test.go

```diff
@@ -1 +1,210 @@
+}
+
+func (s *ClientTestSuite) TearDownTest() {
+	s.client.Close()
+	s.wl.Stop()
+}
+
+func (s *ClientTestSuite) TestFetchX509SVID() {
+	svid, err := s.client.FetchX509SVID(context.TODO())
+	require.NotEmpty(s.T(), svid)
+	require.Nil(s.T(), err)
+}
+
+func (s *ClientTestSuite) TestFetchX509SVIDs() {
+	svid, err := s.client.FetchX509SVIDs(context.TODO())
+	require.Len(s.T(), svid, 2)
+	require.Nil(s.T(), err)
+}
+
+func (s *ClientTestSuite) TestFetchX509Bundles() {
+	svid, err := s.client.FetchX509Bundles(context.TODO())
+	require.NotEmpty(s.T(), svid)
+	require.Nil(s.T(), err)
+}
+
+func (s *ClientTestSuite) TestFetchX509Context() {
+	svid, err := s.client.FetchX509Context(context.TODO())
+	require.Nil(s.T(), err)
+	require.NotEmpty(s.T(), svid)
+	require.Len(s.T(), svid.SVIDs, 2)
+}
+
+func (s *ClientTestSuite) TestWatchX509Context() {
+	w := newTestWatcher(s.T())
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		_ = s.client.WatchX509Context(ctx, w)
+	}()
+
+	w.WaitForUpdates(1)
+	resp := s.makeX509SVIDResponse([]string{"spiffe://example.org/zod"})
+	s.wl.SetX509SVIDResponse(resp)
+
+	w.WaitForUpdates(1)
+	assert.Len(s.T(), w.Errors, 0)
+	assert.Len(s.T(), w.X509SVIDs, 2)
+}
+
+func (s *ClientTestSuite) TestFetchJWTSVID() {
+	spiffeID, _ := spiffeid.New("spiffe://example.org", "mytoken")
+	params := jwtsvid.Params{
+		Subject:        spiffeID,
+		Audience:       "spiffe://example.org/target",
+		ExtraAudiences: []string{"spiffe://example.org/extraAudience"},
+	}
+	svid, err := s.client.FetchJWTSVID(context.TODO(), params)
+
+	require.Nil(s.T(), err)
+	require.NotEmpty(s.T(), svid)
+}
+
+func (s *ClientTestSuite) TestFetchJWTBundles() {
+	spiffeID, _ := spiffeid.New("spiffe://example.org")
+	bundles, err := s.client.FetchJWTBundles(context.TODO())
+	require.Nil(s.T(), err)
+	require.True(s.T(), bundles.Has(spiffeID.TrustDomain()))
+}
+
+func (s *ClientTestSuite) TestWatchJWTBundles() {
+	w := newTestWatcher(s.T())
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		_ = s.client.WatchJWTBundles(ctx, w)
+	}()
+
+	w.WaitForUpdates(1)
+
+	assert.Len(s.T(), w.Errors, 0)
+	assert.Len(s.T(), w.JwtBundles, 1)
+}
+
+func (s *ClientTestSuite) makeX509SVIDResponse(spiffeIDs []string) *test.X509SVIDResponse {
+	svids := []test.X509SVID{}
+	for _, id := range spiffeIDs {
+		svid, key := s.ca.CreateX509SVID(id)
+		svids = append(svids, test.X509SVID{CertChain: svid, Key: key})
+	}
+	return &test.X509SVIDResponse{
+		Bundle: s.ca.Roots(),
+		SVIDs:  svids,
+	}
+}
+
+func (s *ClientTestSuite) makeJWTSVIDResponse(spiffeID string, audience []string) *workload.JWTSVIDResponse {
+	token := s.ca.CreateJWTSVID(spiffeID, audience)
+	svids := []*workload.JWTSVID{
+		{
+			SpiffeId: spiffeID,
+			Svid:     token,
+		},
+	}
+	return &workload.JWTSVIDResponse{
+		Svids: svids}
+}
+
+type testWatcher struct {
+	t            *testing.T
+	X509SVIDs    []*X509Context
```

Comment:
Since the watcher could be mutating these from another goroutine, it's probably better to make them unexpected, protected by a mutex, with getter methods for the test to observe them.